src/devices/src/virtio/vsock/device.rs

```diff
@@ -213,6 +215,32 @@ where
 
         have_used
     }
+
+    // Send TRANSPORT_RESET_EVENT to driver. According to specs, the driver shuts down established
+    // connections and the guest_cid configuration field is fetched again. Existing listen sockets remain
+    // but their CID is updated to reflect the current guest_cid.
+    pub fn send_transport_reset_event(&mut self) -> result::Result<(), DeviceError> {
+        let mem = match self.device_state {
+            DeviceState::Activated(ref mem) => mem,
+            // This should never happen, it's been already validated in the caller function.
+            DeviceState::Inactive => unreachable!(),
+        };
+
+        if let Some(head) = self.queues[EVQ_INDEX].pop(mem) {
```

Comment:
I think we also need to account for the situation when there are no available buffers.

Something like:
```
        let head_desc =self.queues[EVQ_INDEX].pop(mem).ok_or_else(|| {
            IncrementSomeMetric();
            SomeError::EmptyQueue
        })?;
```